samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -146,6 +146,11 @@ public final void processAsync(IncomingMessageEnvelope ime, MessageCollector col
               callback.complete();
             }
           });
+        } else {
+          // If InputOperator is not found in the operator graph for a given SystemStream, throw an exception else the
+          // job will timeout due to async task callback timeout (TaskCallbackTimeoutException)
+          callback.failure(new SamzaException(
```

Comment:
Can we log the error as well similar to the pattern in the rest of the class before invoking `callback.failure(...)`? Callback failures are asynchronous and other exceptions can happen as well and hence, having a log helps drafting timelines during debugging.